sds_data_manager/lambda_code/IAlirtCode/ialirt_db_query_api_formatted.py

```diff
@@ -0,0 +1,196 @@
+"""I-ALiRT Database Query lambda."""
+
+import json
+import logging
+import os
+from decimal import Decimal
+
+import boto3
+from boto3.dynamodb.conditions import Key
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def process_item_types(item: dict) -> dict:
+    """Convert Decimal values to int/float for known fields.
+
+    Parameters
+    ----------
+    item : dict
+        The item in the dictionary.
+
+    Returns
+    -------
+    result : dict
+        Properly formatted parameters.
+    """
+    result = {}
+
+    for key, value in item.items():
+        # Vectors fields
+        if isinstance(value, list):
+            result[key] = [int(v) if v % 1 == 0 else round(float(v), 3) for v in value]
+
+        # Scalar fields
+        elif isinstance(value, Decimal):
+            result[key] = int(value) if value % 1 == 0 else round(float(value), 3)
+
+        else:
+            result[key] = value
+
+    return result
+
+
+def lambda_handler(event, context):  # noqa: PLR0912, PLR0915
+    """Create metadata and add it to the database.
+
+    This function is an event handler for s3 ingest bucket.
+    It is also used to ingest data to the DynamoDB table.
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+    context : LambdaContext
+        This object provides methods and properties that provide
+        information about the invocation, function,
+        and runtime environment.
+
+    """
+    table_name = os.environ.get("ALGORITHM_TABLE")
+    region = os.environ.get("AWS_DEFAULT_REGION", "us-west-2")
+    dynamodb = boto3.resource("dynamodb", region_name=region)
+    table = dynamodb.Table(table_name)
+
+    logger.info(f"Received event: {json.dumps(event)}")
+    params = event.get("queryStringParameters", {})
+
+    if not params:
+        return {
+            "statusCode": 400,
+            "body": json.dumps({"message": "No query parameters provided"}),
+        }
+
+    key_expr = Key("apid").eq(478)
+    query_kwargs = {"KeyConditionExpression": key_expr}
+
+    allowed_params = {
+        "met_start",
+        "met_end",
+        "met_in_utc_start",
+        "met_in_utc_end",
+        "last_modified_start",
+        "last_modified_end",
+    }
```

Comment:
Remove MET, that is an implementation detail and we don't necessarily want to leak that to the public. A public person only wants to enter with "human time" in mind. Then behind the scenes you can do all of the conversions necessary to make it into a MET if that is what is in the DB, but a user doesn't need to know the mapping of the DB, the API should be a human interface not a DB interface.
```suggestion
    allowed_params = {
        "start_time",
        "end_time",
        "last_modified_start",
        "last_modified_end",
    }
```